src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -369,6 +373,36 @@ func (r *Cluster) validatePandaproxyListeners() field.ErrorList {
 	return allErrs
 }
 
+func (r *Cluster) validateSchemaRegistryListener() field.ErrorList {
+	var allErrs field.ErrorList
+	if r.IsSchemaRegistryExternallyAvailable() {
+		kafkaExternal := r.ExternalListener()
+		schemaRegistry := r.Spec.Configuration.SchemaRegistry
+		schemaRegistryFieldPath := field.NewPath("spec").Child("configuration").Child("schemaRegistry")
+
+		if kafkaExternal == nil || !kafkaExternal.External.Enabled {
+			allErrs = append(allErrs,
+				field.Invalid(schemaRegistryFieldPath,
+					r.Spec.Configuration.SchemaRegistry,
+					"cannot have a schema registry external listener without a kafka external listener"))
+		}
+		if kafkaExternal == nil && schemaRegistry.External.Subdomain != "" {
+			allErrs = append(allErrs,
+				field.Invalid(schemaRegistryFieldPath,
+					r.Spec.Configuration.SchemaRegistry,
+					"kafka external listener is empty but must specify the same sudomain as that of the external schema registry"))
```

Comment:
I don't see us validating that the subdomain is the same 🤔 just that it's non-empty